elyra/cli/pipeline_app.py

```diff
@@ -311,5 +310,69 @@ def run(pipeline_path):
     print_banner("Elyra Pipeline Local Run Complete")
 
 
+@click.command()
+@click.option('--json',
+              'json_option',
+              is_flag=True,
+              required=False,
+              help='Returns pipeline file in a machine-readable JSON format')
+@click.argument('pipeline_path')
+def describe(json_option, pipeline_path):
+    """
+    Display pipeline description in a human readable format
+    """
+
+    _validate_pipeline_file_extension(pipeline_path)
+
+    pipeline_definition = _preprocess_pipeline(pipeline_path)
+
+    indent_length = 4
+    for current_pipeline in pipeline_definition["pipelines"]:
+        pipeline_keys = ["name", "description", "type", "version", "nodes", "dependencies"]
+
+        iter_keys = {"dependencies"}
+
+        describe_dict = OrderedDict()
+
+        properties = current_pipeline["app_data"]["properties"]
+
+        pipeline_data = current_pipeline["app_data"]
+
+        # If the name is actually "None", it will seem as if there is no name
+        # The same can be said for all fields
+        describe_dict["name"] = str(properties.get("name", "None"))
+
+        describe_dict["description"] = str(properties.get("description", "None"))
+
+        describe_dict["type"] = str(properties.get("runtime", "None"))
+
+        describe_dict["version"] = str(pipeline_data.get("version", "None"))
+
+        describe_dict["nodes"] = str(len(current_pipeline.get("nodes", [])))
+
+        describe_dict["dependencies"] = set()
+        for node in current_pipeline["nodes"]:
+            if "dependencies" in node["app_data"]["component_parameters"]:
+                for dependency in node["app_data"]["component_parameters"].get("dependencies", []):
+                    describe_dict["dependencies"].add(f"{dependency}")
+
+        if not json_option:
+            for key in pipeline_keys:
+                if key in iter_keys:
+                    click.echo(f"{key.title()}:")
+                    if describe_dict.get(key, set()) == set():
+                        click.echo(f"{' ' * indent_length}None")
+                    else:
+                        for item in describe_dict.get(key, ["None"]):
+                            click.echo(f"{' ' * indent_length}{item}")
+                else:
+                    click.echo(f"{key.title()}: {describe_dict.get(key, 'None')}")
+        else:
+            for key in iter_keys:
+                describe_dict[key] = list(describe_dict[key])
+            click.echo(json.dumps(describe_dict, indent=indent_length))
```

Comment:
The JSON dump currently encodes all values as string, even though some of them are always numeric (e.g. `nodes` or `version`). 

```
$ elyra-pipeline describe no_comment.pipeline --json
{
    "name": "drop_even_rows",
    "description": "None",
    "type": "Kubeflow Pipelines",
    "version": "4",
    "nodes": "1",
    "dependencies": []
}
```

Also note that `None/not set` (the value of `description` in above example) cannot be programmatically distinguished from a  user provided value "`None`". Therefore it would be better to encode `None/not set` as an empty string `""` to make that distinction possible.

Reply:
My most recent changes make it so that the new output for your example will be:
```
$ elyra-pipeline describe no_comment.pipeline --json
{
    "name": "drop_even_rows",,
    "type": "Kubeflow Pipelines",
    "version": "4",
    "nodes": "1",
    "dependencies": []
}
```
The JSON output won't even contain the keys that would be None/null as long as the original `.pipeline` didn't have the key either. This way there will be a difference in output between a pipeline with `""` as the description and the pipelines that have no `"description"` key. 
Does this sound good?

Reply:
That would be another option. Two comments though:

```
"name": "drop_even_rows",,
```

Should not end with two `,,`

```
"dependencies": []
```

For the sake of consistency the same approach would have to be taken for all _optional_ properties, such as the file dependencies.

Reply:
> ```
> "name": "drop_even_rows",,
> ```
> 
> Should not end with two `,,`

Yeah, that was a typo. My bad. It should be one comma in the CLI.

I've just pushed the change that makes sure any empty iterable values (like the list of dependencies) won't be in the JSON output.